examples/dnsdoctor/dnsdoctor.lua

```diff
@@ -0,0 +1,68 @@
+--
+-- SPDX-FileCopyrightText: (c) 2024 Mohammad Shehar Yaar Tausif <sheharyaar48@gmail.com>
+-- SPDX-License-Identifier: MIT OR GPL-2.0-only
+--
+
+-- DNS Doctoring : Rewrite DNS type A record to a private address for local clients
+
+local xt = require("xtable")
+local linux = require("linux")
+local string = require("string")
+local action = xt.action
+local family = xt.family
+
+local udp = 0x11
+local dns = 0x35
+
+local function nop() end
+
+local function get_domain(skb, off)
+	local i = tostring(skb):sub(off+1):find('\0')
+	return skb:getstring(off, i - 1), i
```

Comment:
```suggestion
	local _, nameoff, name = skb:getstring(off):find("([^\0]*)")
	return name, nameoff
```

I think this version is better, because we convert `skb + off` to a string , instead of starting it from the very beginning.. also, we don't create another string afterwards.. we just let `find()` to capture our pattern.. moreover, it looks a bit clearer what we are really doing here.. what do you think?